bundles/org.eclipse.equinox.preferences/src/org/eclipse/core/internal/preferences/PreferencesService.java

```diff
@@ -1097,11 +1099,25 @@ public boolean visit(IEclipsePreferences node) {
 		return result;
 	}
 
+	/*
+	 * Prepend the given scope to the default search order to use when there is
+	 * nothing else set. Clients should not call this method because it is in an
+	 * internal class and has been created solely for use by the
+	 * org.eclipse.core.resources bundle in response to this bug:
+	 * https://bugs.eclipse.org/330320
+	 */
+	public void prependScopeToDefaultDefaultLookupOrder(String firstScope) {
+		List<String> scopes = new ArrayList<>(DEFAULT_DEFAULT_LOOKUP_ORDER);
+		scopes.add(0, firstScope);
+		DEFAULT_DEFAULT_LOOKUP_ORDER = List.copyOf(scopes);
+	}
+
 	/*
 	 * Return the default search lookup order for when nothing is set.
 	 */
+	@Deprecated(forRemoval = true)
```

Comment:
Once https://github.com/eclipse-platform/eclipse.platform/pull/972 is submitted, the deprecated method can be removed.